### ISSUE TITLE
emitc buffer_ops.h generates possible uninitialized code

### ISSUE THREAD

**Issue by hcindyl** [iree-org/iree#14540]
This is a follow-up issue for https://github.com/openxla/iree/issues/14507

The code in question is the `iree_c_module` generated code from https://github.com/openxla/iree/blob/main/runtime/src/iree/vm/test/buffer_ops.mlir built with GCC

The example of warnings is shown in https://github.com/openxla/iree/actions/runs/5723058158/job/15507036727
```

/work/build-gcc/runtime/src/iree/vm/test/emitc/buffer_ops.h:6825:6: warning: 'v29' may be used uninitialized in this function [-Wmaybe-uninitialized]
 6825 |   if (v37) {
      |      ^
/work/build-gcc/runtime/src/iree/vm/test/emitc/buffer_ops.h:6667:11: note: 'v29' was declared here
 6667 |   int32_t v29;
      |           ^~~
/work/build-gcc/runtime/src/iree/vm/test/emitc/buffer_ops.h:6859:6: warning: 'v43' may be used uninitialized in this function [-Wmaybe-uninitialized]
 6859 |   if (v51) {
      |      ^
/work/build-gcc/runtime/src/iree/vm/test/emitc/buffer_ops.h:6681:11: note: 'v43' was declared here
 6681 |   int32_t v43;
      |           ^~~
/work/build-gcc/runtime/src/iree/vm/test/emitc/buffer_ops.h:6893:6: warning: 'v57' may be used uninitialized in this function [-Wmaybe-uninitialized]
 6893 |   if (v65) {
      |      ^
/work/build-gcc/runtime/src/iree/vm/test/emitc/buffer_ops.h:6695:11: note: 'v57' was declared here
 6695 |   int32_t v57;
      |           ^~~
...
```

**Comment by hcindyl** [comment]
warning temporarily silenced with https://github.com/openxla/iree/pull/14537

**Comment by simon-camp** [comment]
Hi @hcindyl, #14589 tries to work around the issue. I wasn't able to reproduce the warnings with gcc 11.4.0, so I will try to build with version 9.4.0 tomorrow.

**Comment by hcindyl** [comment]
Thanks @simon-camp! IREE CI uses GCC9 (see [discussion](https://github.com/openxla/iree/issues/6959)) I am not sure if there are plans to bump the gcc version. @ScottTodd thoughts?

Since you have started working on the issue, I will close my PR.

**Comment by simon-camp** [comment]
This should be fixed with #14589 being landed.
